flink-runtime/src/main/java/org/apache/flink/runtime/state/metrics/LatencyTrackingStateFactory.java

```diff
@@ -34,35 +35,47 @@
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-/** Factory to create {@link AbstractLatencyTrackState}. */
+/** Factory to create {@link AbstractMetricsTrackState}. */
 public class LatencyTrackingStateFactory<
         K, N, V, S extends State, IS extends InternalKvState<K, N, ?>> {
 
     private final InternalKvState<K, N, ?> kvState;
     private final StateDescriptor<S, V> stateDescriptor;
     private final LatencyTrackingStateConfig latencyTrackingStateConfig;
+    private final SizeTrackingStateConfig sizeTrackingStateConfig;
     private final Map<StateDescriptor.Type, SupplierWithException<IS, Exception>> stateFactories;
+    private final KeyedStateBackend keyedStateBackend;
 
     private LatencyTrackingStateFactory(
             InternalKvState<K, N, ?> kvState,
+            KeyedStateBackend<K> keyedStateBackend,
             StateDescriptor<S, V> stateDescriptor,
-            LatencyTrackingStateConfig latencyTrackingStateConfig) {
+            LatencyTrackingStateConfig latencyTrackingStateConfig,
+            SizeTrackingStateConfig sizeTrackingStateConfig) {
         this.kvState = kvState;
+        this.keyedStateBackend = keyedStateBackend;
         this.stateDescriptor = stateDescriptor;
         this.latencyTrackingStateConfig = latencyTrackingStateConfig;
+        this.sizeTrackingStateConfig = sizeTrackingStateConfig;
         this.stateFactories = createStateFactories();
     }
 
     /** Create latency tracking state if enabled. */
     public static <K, N, V, S extends State>
             InternalKvState<K, N, ?> createStateAndWrapWithLatencyTrackingIfEnabled(
                     InternalKvState<K, N, ?> kvState,
+                    KeyedStateBackend<K> keyedStateBackend,
                     StateDescriptor<S, V> stateDescriptor,
-                    LatencyTrackingStateConfig latencyTrackingStateConfig)
+                    LatencyTrackingStateConfig latencyTrackingStateConfig,
+                    SizeTrackingStateConfig sizeTrackingStateConfig)
                     throws Exception {
-        if (latencyTrackingStateConfig.isEnabled()) {
+        if (latencyTrackingStateConfig.isEnabled() || sizeTrackingStateConfig.isEnabled()) {
```

Comment:
It seems the `LatencyTrackingStateFactory` is still used, why this is not renamed?